src/sentry/static/sentry/app/components/lazyLoad.jsx

```diff
@@ -1,35 +1,101 @@
-import React from 'react';
 import PropTypes from 'prop-types';
+import Raven from 'raven-js';
+import React from 'react';
 
+import {t} from '../locale';
+import LoadingError from './loadingError';
 import LoadingIndicator from '../components/loadingIndicator';
 
 class LazyLoad extends React.Component {
   static propTypes = {
     hideBusy: PropTypes.bool,
+    hideError: PropTypes.bool,
     /**
-     * specifically needs to be a thenable
+     * Function that returns a promise of a React.Component
+     * Alias property for `getComponent` to maintain compatibility
      */
     component: PropTypes.func,
+
+    /**
+     * Function that returns a promise of a React.Component
+     */
+    getComponent: PropTypes.func,
+
+    /**
+     * Also accepts a route object from react-router that has a `getComponent` property
+     */
+    route: PropTypes.shape({
+      getComponent: PropTypes.func,
```

Comment:
This is `getComponent` because `component` is an existing prop for a react-router `Route`